src/scanner.c

```diff
@@ -0,0 +1,205 @@
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+#include <tree_sitter/parser.h>
+
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
+
+enum TokenType {
+    HERE_STRING_BODY,
+};
+
+typedef struct {
+    uint32_t len;
+    uint32_t cap;
+    char *data;
+} String;
+
+static String string_new() {
+    return (String){.cap = 16, .len = 0, .data = calloc(1, sizeof(char) * 17)};
```

Comment:
consider write a comment that points out why we have 17 not 16.
Please process the case that `calloc` fails and return `NULL`. I think we can just `exit(EXIT_FAILURE);` in this case.